pkg/controller/controller.go

```diff
@@ -68,28 +68,48 @@ func NewManagerForRestConfig(conf *config.Config, rc *rest.Config) (ctrl.Manager
 	}
 	m.GetLogger().V(2).Info("using namespace: " + conf.NS)
 
-	if err = ingress.NewIngressControllerReconciler(m, manifests.IngressControllerResources(conf, deploy)); err != nil {
+	if err := dns.NewExternalDns(m, conf, deploy); err != nil {
 		return nil, err
 	}
-	if err = ingress.NewConcurrencyWatchdog(m, conf); err != nil {
+
+	nginxConfigs, err := nginx.New(m, conf, deploy)
+	if err != nil {
 		return nil, err
 	}
-	if err = keyvault.NewIngressSecretProviderClassReconciler(m, conf); err != nil {
+
+	watchdogTargets := make([]*ingress.WatchdogTarget, 0)
+	for _, nginxConfig := range nginxConfigs {
+		watchdogTargets = append(watchdogTargets, &ingress.WatchdogTarget{
+			ScrapeFn:    ingress.NginxScrapeFn,
+			LabelGetter: nginxConfig,
+		})
+	}
+	if err := ingress.NewConcurrencyWatchdog(m, conf, watchdogTargets); err != nil {
 		return nil, err
 	}
-	if err = keyvault.NewPlaceholderPodController(m, conf); err != nil {
+
+	ingressManagers := make([]keyvault.IngressManager, 0)
+	for _, nginxConfig := range nginxConfigs {
+		ingressManagers = append(ingressManagers, nginxConfig)
+	}
+	if err := keyvault.NewIngressSecretProviderClassReconciler(m, conf, ingressManagers); err != nil {
 		return nil, err
 	}
-	if err = keyvault.NewEventMirror(m, conf); err != nil {
+	if err := keyvault.NewPlaceholderPodController(m, conf, ingressManagers); err != nil {
```

Comment:
more of a question than feedback - why instantiate a new error object here but then overwrite it in the next statement?

Reply:
it doesn't really matter. It's not overwritten because := is scoped to whatever block it's in. The err is exclusive to that if statement.

Reply:
I do tend to prefer := because I think block scoping things is really good. It limits where you'd have to look and reason about in the case of a bug.